src/framework/components/button/system.js

```diff
@@ -0,0 +1,57 @@
+pc.extend(pc, function () {
+    var _schema = [
+        'enabled',
+        'active',
+        'imageEntity',
+        { name: 'hitPadding', type: 'vec4' },
+        'transitionMode',
+        { name: 'hoverTint', type: 'rgba' },
+        { name: 'pressedTint', type: 'rgba' },
+        { name: 'inactiveTint', type: 'rgba' },
+        'fadeDuration',
+        'hoverSpriteAsset',
+        'hoverSpriteFrame',
+        'pressedSpriteAsset',
+        'pressedSpriteFrame',
+        'inactiveSpriteAsset',
+        'inactiveSpriteFrame'
+    ];
+
+    /**
+     * @private
+     * @name pc.ButtonComponentSystem
+     * @description Create a new ButtonComponentSystem
+     * @classdesc Manages creation of {@link pc.ButtonComponent}s.
+     * @param {pc.Application} app The application
+     * @extends pc.ComponentSystem
+     */
+    var ButtonComponentSystem = function ButtonComponentSystem(app) {
```

Comment:
Do you also need to handle cloneComponent? I guess when you clone the Button you want it to reference its child entities not the original entities.

Reply:
Ah, you mean in the case where someone clones it via the JS API rather than via the editor? Right, yes I guess we do.

Out of interest, is there a reason that duplicating via the editor doesn't make use of engine methods like `cloneComponent`? Seems odd to have to implement things like the duplicated entity reference resolution in two places.

Reply:
Well this is when someone does this at runtime not in the Editor. The Editor handles duplicating on its own because the Editor doesn't normally rely on the engine to do stuff - it just notifies the engine of changes. When you are duplicating in the Editor you are essentially just creating a new Entity in the engine not cloning it.

Reply:
Sure, I understand that – what I'm saying is that if the editor did rely on engine methods like `createComponent` rather than implementing the same processes separately, we could just concentrate on implementing them in one place. There'd be less code to write/maintain in the editor, and bugs or inconsistencies that arise from differing behaviour in the editor vs the engine could not occur.

For something like duplication, the end goal is the same: the user wants to end up with a copy of a subtree with exactly the same properties as the original, regardless of whether they did that via the JS API or via the editor. As such it seems to make sense to just write it once.

Reply:
Well it was an early decision I guess to implement duplicate that way - I think the Editor doesn't know if something is duplicated or not, it's just an entities:add message which creates an engine Entity as well.

Reply:
Ok – anyway I guess this still needs doing :) will add to the list.

- [ ] Implement `cloneComponent()`

Reply:
Implemented here: https://github.com/playcanvas/engine/pull/1200